fastly/service_authorization.go

```diff
@@ -0,0 +1,127 @@
+package fastly
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/google/jsonapi"
+)
+
+type SAUser struct {
+	ID string `jsonapi:"primary,user"`
+}
+
+type SAService struct {
+	ID string `jsonapi:"primary,service"`
+}
```

Comment:
It would be great if the `User` struct already was `jsonapi`'d and I could re-use it directly. But the way it currently stands, the `User` struct still has the `mapstructure` tags which makes it incompatible with the `jsonapi` exposed by service authorizations (if I understood it correctly).

Alternatively to creating these structs I could add the relevant `jsonapi` to the [User](https://github.com/fastly/go-fastly/blob/main/fastly/user.go#L12) and [Service](https://github.com/fastly/go-fastly/blob/main/fastly/service.go#L15) structs and re-use them, but I'm not sure mixing tags like this is the best course. Thoughts?

Reply:
Yeah I would tend to agree and opt for not overlapping the tags just in case it causes any unexpected issues. 👍🏻 

So I think recreating the structs is fine.